sdk/identity/Azure.Identity/src/DeviceCodeCredential.cs

```diff
@@ -17,44 +17,32 @@ namespace Azure.Identity
     /// </summary>
     public class DeviceCodeCredential : TokenCredential
     {
-        private readonly MsalPublicClient _client = null;
-        private readonly CredentialPipeline _pipeline;
-        private AuthenticationRecord _record = null;
-        private readonly string _clientId;
-        private readonly Func<DeviceCodeInfo, CancellationToken, Task> _deviceCodeCallback;
-        private bool _disableAutomaticAuthentication = false;
+        internal MsalPublicClient Client { get; }
+        internal string ClientId { get; }
+        internal bool DisableAutomaticAuthentication { get; }
+        internal AuthenticationRecord Record { get; private set; }
+        internal Func<DeviceCodeInfo, CancellationToken, Task> DeviceCodeCallback { get; }
+        internal CredentialPipeline Pipeline { get; }
+
         private const string AuthenticationRequiredMessage = "Interactive authentication is needed to acquire token. Call Authenticate to initiate the device code authentication.";
         private const string NoDefaultScopeMessage = "Authenticating in this environment requires specifying a TokenRequestContext.";
 
         /// <summary>
-        /// Creates a new <see cref="DeviceCodeCredential"/>, which will authenticate users using the device code flow, printing the device code message to stdout.
+        /// Creates a new <see cref="DeviceCodeCredential"/>, which will authenticate users using the device code flow.
         /// </summary>
         public DeviceCodeCredential() :
-            this(DefaultDeviceCodeHandler, null, null, null, null)
+            this(DefaultDeviceCodeHandler, null, Constants.DeveloperSignOnClientId, null, null)
```

Comment:
Nit - Maybe just call `this(null)` so the defaults are provided just once in the overload below?